### ISSUE TITLE
Example code doesn't run with Python 3.12

### ISSUE THREAD

**Issue by simonprickett** [redislabs-training/ru101#16]
It looks like with Python 3.12 that the `distutils` package that redis-py uses isn't present.  This was raised by a student on Discord:

```
➜ python redisu/ru101/uc01-faceted-search/search.py
Traceback (most recent call last):
  File "/home/nbarreiro/Repos/redis-university/ru101/redisu/ru101/uc01-faceted-search/search.py", line 4, in <module>
    from redis import Redis
  File "/home/nbarreiro/Repos/redis-university/ru101/venv/lib/python3.12/site-packages/redis/__init__.py", line 1, in <module>
    from redis.client import Redis, StrictRedis
  File "/home/nbarreiro/Repos/redis-university/ru101/venv/lib/python3.12/site-packages/redis/client.py", line 12, in <module>
    from redis.connection import (ConnectionPool, UnixDomainSocketConnection,
  File "/home/nbarreiro/Repos/redis-university/ru101/venv/lib/python3.12/site-packages/redis/connection.py", line 2, in <module>
    from distutils.version import StrictVersion
ModuleNotFoundError: No module named 'distutils'
```

Try upgrading the version of redis-py used in this course: https://github.com/redislabs-training/ru101/blob/main/requirements.txt - currently quite old (3.0.1).

**Comment by simonprickett** [comment]
See also https://stackoverflow.com/questions/77233855/why-did-i-got-an-error-modulenotfounderror-no-module-named-distutils

**Comment by simonprickett** [comment]
I've been able to reproduce this with Python 3.12.0rc2 which is what's available via pyenv at the moment.

```
(venv) redisu $ python utils/dumpload.py load ru101/data/ru101.json
Traceback (most recent call last):
  File "/Users/simonprickett/source/github/ru101/redisu/utils/dumpload.py", line 8, in <module>
    from redis import StrictRedis
  File "/Users/simonprickett/source/github/ru101/venv/lib/python3.12/site-packages/redis/__init__.py", line 1, in <module>
    from redis.client import Redis, StrictRedis
  File "/Users/simonprickett/source/github/ru101/venv/lib/python3.12/site-packages/redis/client.py", line 12, in <module>
    from redis.connection import (ConnectionPool, UnixDomainSocketConnection,
  File "/Users/simonprickett/source/github/ru101/venv/lib/python3.12/site-packages/redis/connection.py", line 2, in <module>
    from distutils.version import StrictVersion
ModuleNotFoundError: No module named 'distutils'
(venv) redisu $ python --versioj
unknown option --versioj
usage: python [option] ... [-c cmd | -m mod | file | -] [arg] ...
Try `python -h' for more information.
(venv) redisu $ python --version
Python 3.12.0rc2
```

**Comment by simonprickett** [comment]
I'm moving this to redis-py 5.0.1 which will fix this issue.  It does introduce some deprecation warnings that I'll fix at the same time.  Example (I'll log all of these as I test each course script):

```
(venv) redisu $ python utils/dumpload.py load ru101/data/ru101.json
/Users/simonprickett/source/github/ru101/redisu/utils/dumpload.py:76: DeprecationWarning: Pipeline.hmset() is deprecated. Use Pipeline.hset() instead.
  p.hmset(obj['k'], obj['v'])
total keys loaded: 14328
```

**Comment by simonprickett** [comment]
Found an issue with `uc05` where the `GEOADD` signature has changed:

```
(venv) uc05-finding-venues $ python finding_venues.py

==Test 1 - geo searches around a venue
Traceback (most recent call last):
  File "/Users/simonprickett/source/github/ru101/redisu/ru101/uc05-finding-venues/finding_venues.py", line 163, in <module>
    main()
  File "/Users/simonprickett/source/github/ru101/redisu/ru101/uc05-finding-venues/finding_venues.py", line 157, in main
    test_venue_search()
  File "/Users/simonprickett/source/github/ru101/redisu/ru101/uc05-finding-venues/finding_venues.py", line 69, in test_venue_search
    create_venue(olympic_stadium)
  File "/Users/simonprickett/source/github/ru101/redisu/ru101/uc05-finding-venues/finding_venues.py", line 64, in create_venue
    redis.geoadd(key, venue['geo']['long'], venue['geo']['lat'], venue['venue'])
  File "/Users/simonprickett/source/github/ru101/venv/lib/python3.12/site-packages/redis/commands/core.py", line 5438, in geoadd
    raise DataError("GEOADD allows either 'nx' or 'xx', not both")
redis.exceptions.DataError: GEOADD allows either 'nx' or 'xx', not both
```

**Comment by simonprickett** [comment]
Closed by #17
